### PR TITLE
MT-2843 checkout flow

### PR DESCRIPTION
**Pull request by yulia-dnistrian**
Description:

Linting:

- [x] No linting errors

Tests:

- [ ] E2E tests
- [x] Manual tests
- [ ] Accessibility tests (no new react-axe errors in console)

Documentation:

- [x] Requires documentation updates
- [ ] Requires Storybook component updates

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,12 +3,13 @@
   "version": "1.1.0",
   "private": true,
   "dependencies": {
-    "@moltin/sdk": "^4.0.0",
+    "@moltin/sdk": "^4.0.2",
```
Comment (shaunmaharaj) [inline package.json:6]: thank you!

File: src/config.ts [hunk 2]
```diff
@@ -6,6 +6,8 @@ export const config = {
   maxCompareProducts: 4,
   algoliaAppId: '6MAFUOD23U',
   algoliaApiKey: '7d829396bbf202b30bfe7ad52120bbac',
+  algoliaPlacesAppId: 'plEPUZAA2D2L',
+  algoliaPlacesApiKey: '4c9f0832a65f800e31b0d50f44670b1f',
   algoliaIndexName: 'product',
   compareKeys: [],
   shopkitSrc: 'https://btn.moltin.com',
```
Comment (shaunmaharaj) [inline src/config.ts:13]: We'll need to remove this config (shopkitSrc) as it's not needed anymore with the native checkout.

File: src/AppHeader.tsx [hunk 3]
```diff
@@ -32,10 +43,11 @@ export const AppHeader: React.FC = () => {
           <LanguageDropdown />
         </div>
         <div className="appheader__moltincartcontainer">
-          <span
-            className="moltin-cart-button"
-            key={id}
-          ></span>
+          <button className="epbtn --bordered" onClick={handleCartModal}>
```
Comment (shaunmaharaj) [inline src/AppHeader.tsx:46]: Let's show the cart icon button in mobile like we previously did, to save on space:
image

File: src/AppHeader.tsx [hunk 4]
```diff
@@ -32,10 +43,11 @@ export const AppHeader: React.FC = () => {
...
+            {count}
```
Comment (shaunmaharaj) [inline src/AppHeader.tsx:47]: Let's try replicating the string message from ShopKit (Cart ()items):
image

File: src/CartModal.scss [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+  }
+
+  &__content {
+    position: fixed;
+    top: 0px;
+    bottom: 0px;
+    right: 0px;
+    overflow-y: scroll;
+    height: 100%;
+    width: 500px;
```
Comment (shaunmaharaj) [inline src/CartModal.scss:26]: 500px width is fixed on mobile. Need to adjust this for mobile views so it wraps contents:
image

Compared to ShopKit:
image

File: src/PlacesSuggest.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  type: ['city', 'address'],
+  useDeviceLocation: false,
+  style: false
+};
+
+export const PlacesSuggest: React.FC<PlacesSuggestParams> = (props) => {
+
+  return (
+    <div className="algoliasearch">
+      <div className="algoliasearch__title">Search for your Shipping address</div>
```
Comment (shaunmaharaj) [inline src/PlacesSuggest.tsx:24]: Need to externalize string

File: src/PlacesSuggest.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+      <AlgoliaPlaces
+        id="AlgoliaPlaces"
+        options={options}
+        placeholder="Start typing an address"
```
Comment (shaunmaharaj) [inline src/PlacesSuggest.tsx:29]: Can we externalize this string?

File: src/CartModal.tsx [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+    if(route === "shipping") {
+      setRoute("itemList")
+    } else if (route === "billing") {
+      setRoute("shipping")
+    }
+  };
+
+  const validate = (email: string) => {
+    const expression = /\S+@\S+/;
+
```
Comment (shaunmaharaj) [inline src/CartModal.tsx:98]: remove empty line

File: src/LoginDialog.tsx [hunk 9]
```diff
@@ -4,7 +4,7 @@ import { Link } from 'react-router-dom';
 import Modal from 'react-responsive-modal';
 import { useFormik } from 'formik';
 import { login } from './service';
-import { useCustomerData, useTranslation } from './app-state';
+import {useCartData, useCustomerData, useTranslation} from './app-state';
```
Comment (shaunmaharaj) [inline src/LoginDialog.tsx:7]: missing spaces between braces

File: src/config.ts (hunk 2 above)
Comment (petro97) [reply src/config.ts:13]: @shaunmaharaj, I still using AlgoliaPlaces to autocomplete address search. Isn't it necessary to use Algolia Places here?
https://github.com/elasticpath/epcc-react-pwa-reference-storefront/blob/161b842839ef64b6228d455268bd018ca2bb81a5/src/PlacesSuggest.tsx#L14

Comment (shaunmaharaj) [reply src/config.ts:13]: @petro97 Algolia configs are correct. We need to remove "shopkitSrc" since it's no longer used anymore in our code 😃

Comment (petro97) [reply src/config.ts:13]: Oh sorry, missed it. I'll remove shopkitSrc

File: src/AccountDropdown.tsx [hunk 10]
```diff
@@ -32,6 +33,7 @@ export const AccountDropdown: React.FC = (props) => {
   });
 
   const logout = () => {
+    localStorage.setItem('mcart', createCartIdentifier());
```
Comment (shaunmaharaj) [inline src/AccountDropdown.tsx:36]: @petro97 @yulia-dnistrian I just noticed this. It seems a bit overkill. We need to import moltin/request into the project only for this call? Is this not supported in our SDK?

Comment (petro97) [reply src/AccountDropdown.tsx:41]: @shaunmaharaj I agree, It seems overkill. I removed @moltin/request

File: src/AddressFields.tsx [hunk 11]
```diff
@@ -65,7 +65,7 @@ export const AddressFields: React.FC<CheckoutParams> = (props) => {
   };
 
   const validate = (values:any) => {
-    onSetAddress(values);
+    // onSetAddress(values);
```
Comment (shaunmaharaj) [inline src/AddressFields.tsx:68]: please remove commented code

File: src/CartModal.tsx [hunk 12]
```diff
@@ -171,15 +181,18 @@ export const CartModal: React.FC<CartModalParams> = (props) => {
...
+                {emailError && (
+                  <div className="epform__error">{emailError}</div>
+                )}
               </div>
             )}
             <div className="cartmodal__body">
               <StripeProvider apiKey={config.stripeKey}>
                 <Elements>
-                  <Checkout shippingAddress={shippingAddress} onPayOrder={onPayOrder} />
+                  <Checkout shippingAddress={shippingAddress} onPayOrder={onPayOrder} isDisabled={!billingAddress.last_name || !email || emailError !== ''} />
```
Comment (shaunmaharaj) [inline src/CartModal.tsx:195]: excellent! Thank you for catching this required check

**Pull request merged**